Source/Scene/ClippingPlanesCollection.js

```diff
@@ -0,0 +1,189 @@
+            packedPlane.w = scratchPlane.distance;
+        }
+
+        return array;
+    };
+
+    /**
+     * Duplicates this ClippingPlanesCollection instance.
+     *
+     * @param {ClippingPlanesCollection} [result] The object onto which to store the result.
+     * @returns he modified result parameter or a new ClippingPlanesCollection instance if one was not provided.
+     */
+    ClippingPlanesCollection.prototype.clone = function (result) {
+        if (!defined(result)) {
+            result = new ClippingPlanesCollection();
+        }
+
+        result.planes = Array.from(this.planes);
+        result.enabled = this.enabled;
+        Matrix4.clone(this.transformationMatrix, result.transformationMatrix);
+        result.inclusive = this.inclusive;
+        Color.clone(this.edgeColor, result.edgeColor);
+        result.edgeWidth = this.edgeWidth;
+
+        return result;
+    };
+
+    /**
+     * Determines the type intersection with the planes of this bounding collection and the specified {@link BoundingVolume}.
+     *
+     * @param {BoundingVolume} boundingVolume The volume to determine the intersection with the planes.
+     * @param {Matrix4} [parentTransform] An optional, additional matrix to transform the plane to world coordinates.
+     * @returns {Intersect} {@link Intersect.INSIDE} if the entire volume is on the side of the planes
+     *                      the normal is pointing and should be entirely rendered, {@link Intersect.OUTSIDE}
+     *                      if the entire volume is on the opposite side and should be clipped, and
+     *                      {@link Intersect.INTERSECTING} if the volume intersects the planes.
+     */
+    ClippingPlanesCollection.prototype.computeIntersectionWithBoundingVolume = function (boundingVolume, parentTransform) {
+        var planes = this.planes;
+        var length = planes.length;
+
+        var transformation = this.transformationMatrix;
+        if (defined(parentTransform)) {
+            transformation = Matrix4.multiply(transformation, parentTransform, scratchMatrix);
+        }
+
+        // If the clipping planes are inclusive, the volume must be outside of all planes to be considered completely
+        // clipped. Otherwise, if the volume can be outside any the planes, it is considered completely clipped.
+        // Lastly, if not completely clipped, if any plane is intersecting, more calculations must be performed.
+        var intersection = Intersect.INSIDE;
+        if (this.inclusive && length > 0) {
+            intersection = Intersect.OUTSIDE;
+        }
+        for (var i = 0; i < length; ++i) {
+            var plane = planes[i];
+
+            Plane.transform(plane, transformation, scratchPlane);
+
+            var value = boundingVolume.intersectPlane(scratchPlane);
+            if (value === Intersect.INTERSECTING) {
+                intersection = value;
+            } else if ((this.inclusive && value === Intersect.INSIDE) ||
```

Comment:
This can be made more efficient - just test `this.inclusive` outside of the loop and have two separate loops.

Overkill - and it might even hurt the JS optimizer - the other option is to make a property setter for `inclusive` and then set the function `computeIntersectionWithBoundingVolume` to one of two functions - which completely moves the `if` statement outside of rendering instead of just outside of this loop.